crates/ruff/src/settings/pyproject.rs

```diff
@@ -114,10 +118,19 @@ pub fn find_user_settings_toml() -> Option<PathBuf> {
 pub fn load_options<P: AsRef<Path>>(path: P) -> Result<Options> {
     if path.as_ref().ends_with("pyproject.toml") {
         let pyproject = parse_pyproject_toml(&path)?;
-        Ok(pyproject
+        let mut ruff = pyproject
             .tool
             .and_then(|tool| tool.ruff)
-            .unwrap_or_default())
+            .unwrap_or_default();
+        if ruff.target_version.is_none() {
+            if let Some(project) = pyproject.project {
+                if let Some(requires_python) = project.requires_python {
+                    ruff.target_version =
+                        PythonVersion::get_minimum_supported_version(&requires_python);
+                }
+            }
+        }
+        Ok(ruff)
     } else {
         parse_ruff_toml(path)
```

Comment:
It will not load `requires-python` if `ruff.toml`is used.

Reply:
could you add a `debug!` for that? my idea is that when someone goes "why is ruff not picking this up?" they can run `RUST_LOG=ruff=debug` and get the answer